x/protocolpool/keeper/keeper.go

```diff
@@ -188,13 +199,55 @@ func (k Keeper) SetToDistribute(ctx context.Context, amount sdk.Coins, addr stri
 		return err
 	}
 
+	totalStreamFundsPercentage := math.ZeroInt()
+	err = k.RecipientFundPercentage.Walk(ctx, nil, func(key sdk.AccAddress, value math.Int) (stop bool, err error) {
+		totalStreamFundsPercentage = totalStreamFundsPercentage.Add(value)
+		return false, nil
+	})
+	if totalStreamFundsPercentage.GT(math.NewInt(100)) {
+		return fmt.Errorf("total funds percentage cannot exceed 100")
+	}
+	if err != nil {
+		return err
+	}
+
+	// send streaming funds to the stream module account
+	if err := k.sendFundsToStreamModule(ctx, denom, totalStreamFundsPercentage); err != nil {
+		return err
+	}
+
 	err = k.ToDistribute.Set(ctx, amount.AmountOf(denom))
 	if err != nil {
 		return fmt.Errorf("error while setting ToDistribute: %v", err)
 	}
 	return nil
 }
 
+func (k Keeper) sendFundsToStreamModule(ctx context.Context, denom string, percentage math.Int) error {
```

Comment:
nit: why not just provide `percentage math.LegacyDec` instead of `Int`, which requires an internal cast.

(not a blocker, just a suggestion)

Reply:
we could, but this func is only called in SetToDistribute and there's a check for total percentage in Int. So just leaving the same type to make it easier there.